### PR TITLE
feat: introduce Switch component

### PR DESCRIPTION
**Pull request by vepor**
This Pull Request meets the following criteria:

- [x] Tests have been added/adjusted for my new feature
- [x] New Components are registered in index.js of my project
- [x] New Components has both `.flow` and `d.ts` files and are exprorted in `index.js.flow` and `index.d.ts`

 Storybook: https://orbit-feat-switch-component.surge.sh

### CODE REVIEW

**Review by silvenon — CHANGES_REQUESTED** [review]
Using a checkbox is a neat CSS trick, but it doesn't follow best practices for a11y, instead I suggest:

- making the entire area clickable instead of just the circle, meaning that the focus ring would go around the whole switch as well (right now the cursor turns into `pointer` in areas which aren't clickable anyway)
- (just continuing from previous note) wrapping everything in `<button type="button" role="switch" aria-checked={checked}>` instead of `StyledSwitchButton`
- expose a `title` prop (or whatever you want to call it) which would be rendered inside the button, but visually hidden, so it would describe what the switch does for screen readers
- wrapping the icon into an element with `aria-hidden="true"`
- making sure that Switch can be toggled with the space bar as well as with mouse click

References:

- [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Switch_role)
- [Switch](https://github.com/tailwindlabs/headlessui/blob/main/packages/@headlessui-react/src/components/switch/switch.tsx) from `@headlessui/react`

**Comment by mainframev** [comment]
> Using a checkbox is a neat CSS trick, but it doesn't follow best practices for a11y, instead I suggest:
> 
> * making the entire area clickable instead of just the circle, meaning that the focus ring would go around the whole switch as well (right now the cursor turns into `pointer` in areas which aren't clickable anyway)
> * (just continuing from previous note) wrapping everything in `<button type="button" role="switch" aria-checked={checked}>` instead of `StyledSwitchButton`
> * expose a `title` prop (or whatever you want to call it) which would be rendered inside the button, but visually hidden, so it would describe what the switch does for screen readers
> * wrapping the icon into an element with `aria-hidden="true"`
> * making sure that Switch can be toggled with the space bar as well as with mouse click
> 
> References:
> 
> * [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Switch_role)
> * [Switch](https://github.com/tailwindlabs/headlessui/blob/main/packages/@headlessui-react/src/components/switch/switch.tsx) from `@headlessui/react`

with current implementation is possible to toggle it with a spacebar ;) Thanks for the review, your points make sense, i will apply them

File: packages/orbit-components/src/Switch/index.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+          onChange={!disabled ? onChange : undefined}
+          onFocus={!disabled ? onFocus : undefined}
+          type="checkbox"
+          data-test={dataTest}
+        />
+        <StyledSwitchButton
+          checked={checked}
+          disabled={disabled}
+          hasCustomIcon={!!icon}
+          role="button"
```
Comment (silvenon) [inline packages/orbit-components/src/Switch/index.js:125]: I think you can remove this `role="button"`, considering that there's no event and it's not focusable.

File: packages/orbit-components/src/Switch/__tests__/index.test.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  it("should be disabled", () => {
+    const onChange = jest.fn();
+    render(<Switch checked onChange={onChange} disabled />);
+
+    userEvent.click(screen.getByRole("button"));
+    expect(onChange).not.toHaveBeenCalled();
+    expect(screen.getByRole("switch")).toHaveAttribute("disabled");
+  });
+});
```
Comment (silvenon) [inline packages/orbit-components/src/Switch/__tests__/index.test.js:34]: Pressing <kbd>⮐</kbd> no longer works, could add a test for it as well as space bar?

File: packages/orbit-components/src/Switch/index.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  position: absolute;
+  opacity: 0;
+  width: 100%;
+  height: 100%;
+  top: 0;
+  left: 0;
+  margin: 0;
+  padding: 0;
+
+  focus + ${StyledSwitchButton} {
```
Comment (silvenon) [inline packages/orbit-components/src/Switch/index.js:81]:
```suggestion
  &:focus + ${StyledSwitchButton} {
```

**Review by silvenon — CHANGES_REQUESTED** [review]

File: packages/orbit-components/src/Switch/__tests__/index.test.js (hunk 2 above)
Comment (mainframev) [reply packages/orbit-components/src/Switch/__tests__/index.test.js:38]: fixed

**Review by silvenon — DISMISSED** [review]
Everything working correctly now.

**Review by mainframev — DISMISSED** [review]

**Pull request merged**